Feat: 수신된 커피챗 신청 목록 조회, 커피챗 신청 상태 변경 기능 추가

jooyeon0907 opened the pull request:
### 변경사항
<!-- 이 PR에서 어떤점들이 변경되었는지 기술해주세요. 가급적이면 as-is, to-be를 활용해서 작성해주세요.  -->
**AS-IS**
- 멘티가 신청한 커피챗 조회 및 신청 기능만 존재
- 커피챗 신청 상태 변경 기능 없음
- User 엔티티의 PK 타입이 `long`
- `CoffeeChatApplicationResponseDto`에 멘토 이름 필드 없음
- `CoffeeChatReceivedService`에서 `CoffeeChatInfoService`와 `CoffeeChatApplicationService`의 메소드 호출이 불가능 (private 접근 제어자)

**TO-BE**
-  **수신된 커피챗 신청 목록 조회 기능 추가**  
    - `GET api/coffee-chats/applications/received` API 추가 
    - 멘토가 받은 커피챗 신청 목록 조회

-  **커피챗 신청 상태 변경 기능 추가**  
    - `PUT api/coffee-chats/applications/{coffeeChatAppId}/received/status` API 추가  
    - 멘티가 신청한 커피챗을 **멘토가 승인(APPROVED) 또는 거절(REJECTED)할 수 있도록 변경**  
    - **검증 로직**  
        - 커피챗 작성자가 요청한 사용자와 일치하는지 확인  
        - 현재 커피챗 신청 상태가 `PENDING(대기 중)`이 아닐 경우 상태 변경 불가  
        - **UI 방어 로직:** "대기 중" 상태일 때만 "거절/수락" 버튼이 노출됨 → 하지만 API 조작 방지를 위해 백엔드에서도 검증 필요하다고 생각하여 추가

- **기타 리팩토링 및 수정 사항**  
  - `User` 엔티티의 **PK 타입을 `long` → `Long`으로 변경** (nullable 가능)  
  - `CoffeeChatApplicationResponseDto`에 **멘토 이름(`mentorName`) 필드 추가**  
  - `CoffeeChatReceivedService`에서 호출을 위해 `CoffeeChatInfoService`와 `CoffeeChatApplicationService`의 메소드 접근 제어자 **private → protected** 변경  
    - `getCoffeeChatInfoByUserId()`, `getCoffeeChatApplication()`  

-  **커피챗 신청 검증 로직 리팩토링**  
    - 기존 `validateCoffeeChatApplication()`을  `validateCoffeeChatApplicant()`  , `validateCoffeeChatInfo()` 로 분리하여 **다른 서비스에서도 사용 가능하도록 수정**
  

  
###  **추후 적용 예정 사항 정리**

####  **포인트 차감 및 회수 처리**
  - **현재**: 포인트 차감 로직 미적용
  - **예정**: 신청 시 포인트 차감 → 멘토가 거절(REJECTED)하면 포인트 회수

####  **채팅룸 생성 및 활성화 로직 추가**
- **멘토가 승인(APPROVED) 시 채팅룸 생성**
- **커피챗 시작 시간이 되면 채팅방 활성화**
  
#### **커피챗 관련 validate 코드 및 공통 메서드 리팩토링**
- **현재**: 커피챗 관련 validate 로직과 공통 메서드들이 여러 서비스에 분산되어 있음
- **의견**: 커피챗과 관련된 validate 메서드와 공통 기능들을 한 곳에 통합하는 것이 좋을 것 같습니다.
  
### 테스트
<!-- 본 변경사항이 테스트가 되었는지 기술해주세요 --> 
- [ ] 테스트 코드
- [x] API 테스트 
  - **수신된 커피챗 신청 목록 조회**
    - 수신된 커피챗 신청 목록 조회 성공 여부 확인  
    - ![수신 커피챗목록](https://github.com/user-attachments/assets/d836e5d9-88a2-4ba1-bb49-568fa5a55b0b)

    - 예외 처리: 작성된 커피챗 정보가 없는 경우 → `"유저에 해당하는 커피챗을 찾을 수 없습니다."`  
      - ![커피챗정보작성안함](https://github.com/user-attachments/assets/4808c504-137e-4fa7-b714-b285d4600d4e)

  - **커피챗 신청 상태 변경**  
    - 커피챗 신청 상태 변경 성공 여부 확인  
    - 예외 처리  
      - 수신된 커피챗 신청이 없을 경우 → `"커피챗 신청 내역을 찾을 수 없습니다."`  
      - ![신청내역없음2](https://github.com/user-attachments/assets/32022c55-111d-4b4a-91fd-836ddea5d0a4)

      - 커피챗 작성자와 요청자가 다를 경우 → `"사용자가 작성한 커피챗 정보가 아닙니다."`  
      - ![사용자작성자아닌ㅁ](https://github.com/user-attachments/assets/af797a60-29fc-422d-9398-2a6670e0e847)


      - 커피챗 신청이 `PENDING` 상태가 아닌 경우 → `"대기 중 상태가 아니므로 상태를 변경할 수 없습니다."`  
      - ![대기중아님](https://github.com/user-attachments/assets/2d3b353e-b2f2-4ef0-820b-0c8178aa96a3)

jooyeon0907 committed:
fix: 경로 매핑 수정
- @GetMapping, @PutMapping, @DeleteMapping에 '/' 추가 ("{coffeeChatAppId}" -> "/{coffeeChatAppId}")

jooyeon0907 committed:
fix: CoffeeChatApplicationResponseDto 수정
- status 타입 String -> StatusType 수정
- user 이름 필드가 name -> userName 으로 변경된 사항에 맞게 수정

jooyeon0907 committed:
Merge branch 'develop' of github.com:BootTalk-Service/boot-talk-back into feat/coffeechat_received

 Conflicts:
	boottalk/src/main/java/com/icandoit/boottalk/coffeeChat/controller/CoffeeChatApplicationController.java
	boottalk/src/main/java/com/icandoit/boottalk/coffeeChat/dto/CoffeeChatApplicationResponseDto.java

jooyeon0907 committed:
Merge branch 'develop' of github.com:BootTalk-Service/boot-talk-back into feat/coffeechat_received

jooyeon0907 committed:
Merge branch 'develop' of github.com:BootTalk-Service/boot-talk-back into feat/coffeechat_received

jooyeon0907 committed:
fix: User 엔티티 및 DTO 수정
- User 엔티티 PK 타입 long → Long 변경
- CoffeeChatApplicationResponseDto에 멘토 이름 필드 추가

jooyeon0907 committed:
chore: 서비스 메소드 접근 제어자 수정 및 검증 로직 분리
- CoffeeChatReceivedService에서 호출할 수 있도록 기존 private 메소드를 protected로 변경
  - getCoffeeChatInfoByUserId(), getCoffeeChatApplication()
- validateCoffeeChatApplication 메소드를 validateCoffeeChatApplicant, validateCoffeeChatInfo로 분리
  - CoffeeChatReceivedService에서 validateCoffeeChatInfo가 필요하여 분리함

jooyeon0907 committed:
feat: 수신된 커피챗 신청 목록 조회, 커피챗 신청 상태 변경 기능 추가

jooyeon0907 committed:
feat: 누락된 코드 추가

shCHO9801 commented on boottalk/src/main/java/com/icandoit/boottalk/coffeeChat/controller/CoffeeChatReceivedController.java:
```diff
@@ -0,0 +1,49 @@
+    private final CoffeeChatReceivedService coffeeChatReceivedService;
+
+    // 수신된 커피챗 신청 목록 조회 (멘티가 나에게 신청한 커피챗 신청 목록 조회)
+    @GetMapping()
```
괄호는 나중에 뭔가 추가가 될 예정인가요??

jooyeon0907 replied:
앗 잘못 입력되었네요!!

shCHO9801 commented on boottalk/src/main/java/com/icandoit/boottalk/coffeeChat/dto/CoffeeChatAppChangeStatusDto.java:
```diff
@@ -0,0 +1,16 @@
+
+public record CoffeeChatAppChangeStatusDto(
+    @NotNull(message = "커피챗 정보 ID는 필수입니다.")
+    Long coffeeChatInfoId,
```
커피쳇 수정시에 coffeeChatInfoId를 사용을 안하시는거 같은데 한번 확인해주세요!

jooyeon0907 replied:
그렇네요  발견해주셔서 감사합니다!

shCHO9801 reviewed (commented):
전체적으로 멘토가 수신한 커피챗 신청을 조회하고 상태를 변경하는 흐름이 잘 정리되어 있습니다.
몇 가지 작은 코멘트는 참고 부탁드립니다 :)

geonuu09 commented on boottalk/src/main/java/com/icandoit/boottalk/coffeeChat/service/CoffeeChatReceivedService.java:
```diff
@@ -0,0 +1,83 @@
+
+        return coffeeChatApplications.stream()
+            .map(coffeeChatApplication -> CoffeeChatApplicationResponseDto.from(coffeeChatApplication))
+            .collect(Collectors.toList());
```
java 16 이상부터 .collect(Collectors.toList()) -> toList() 사용 가능하다고 합니다!

jooyeon0907 replied:
헉 저는 계속  .collect(Collectors.toList()) 사용했었네요 정보감사합니다! 수정하겠습니다

jooyeon0907 committed:
feat: 리뷰 피드백 반영
- CoffeeChatAppChangeStatusDto, CoffeeChatApplicationUpdateDto 에 사용되지 않는 coffeeChatInfoId 필드 제거
- .collect(Collectors.toList()) -> toList() 수정 (java 16 이상부터 지원되는 기능)
